src/coins.cpp

```diff
@@ -1557,6 +1576,65 @@ CValidationState::Code CCoinsViewCache::IsScTxApplicableToState(const CTransacti
     return CValidationState::Code::OK;
 }
 
+#ifdef ENABLE_ADDRESS_INDEXING
+void CCoinsViewCache::HandleIndexesSidechainEvents(int height, CBlockTreeDB* pblocktree,
+                                                   std::vector<std::pair<CAddressIndexKey, CAddressIndexValue>>& addressIndex,
+                                                   std::vector<std::pair<CAddressUnspentKey, CAddressUnspentValue>>& addressUnspentIndex)
+{
+    if (!HaveSidechainEvents(height))
+        return;
+
+    CSidechainEvents scEvents;
+    GetSidechainEvents(height, scEvents);
+
+    //Handle Ceasing Sidechain
+    for (const uint256& ceasingScId : scEvents.ceasingScs)
+    {
+        CSidechain sidechain;
+        assert(GetSidechain(ceasingScId, sidechain));
+
+        if (sidechain.lastTopQualityCertReferencedEpoch == CScCertificate::EPOCH_NULL) {
+            assert(sidechain.lastTopQualityCertHash.IsNull());
+            continue;
+        }
+
+        CTxIndexValue txIndexVal;
+        assert(pblocktree->ReadTxIndex(sidechain.lastTopQualityCertHash, txIndexVal));
+
+        // Set the lower quality BTs as superseded
+        UpdateBackwardTransferIndexes(sidechain.lastTopQualityCertHash, txIndexVal.txIndex, addressIndex, addressUnspentIndex,
+                                      CCoinsViewCache::flagIndexesUpdateType::SUPERSEDE_CERTIFICATE);
+    }
```

Comment:
Update also txIndex setting the lastTopQualCertHash with certMaturityHeight * -1